lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -192,33 +197,37 @@ def emit_partial_range
         @partial_range = nil
       end
 
-      sig { params(node: T.any(SyntaxTree::Call, SyntaxTree::CommandCall)).void }
+      sig { params(node: T.any(SyntaxTree::CallNode, SyntaxTree::CommandCall)).void }
       def add_call_range(node)
         receiver = T.let(node.receiver, SyntaxTree::Node)
         loop do
           case receiver
-          when SyntaxTree::Call
+          when SyntaxTree::CallNode
             visit(receiver.arguments)
             receiver = receiver.receiver
           when SyntaxTree::MethodAddBlock
             visit(receiver.block)
             receiver = receiver.call
 
-            if receiver.is_a?(SyntaxTree::Call) || receiver.is_a?(SyntaxTree::CommandCall)
+            if receiver.is_a?(SyntaxTree::CallNode) || receiver.is_a?(SyntaxTree::CommandCall)
               receiver = receiver.receiver
             end
           else
             break
           end
         end
 
-        add_lines_range(receiver.location.start_line, node.location.end_line - 1)
+        add_lines_range(receiver.location.start_line, node.location.end_line - 1) if receiver
 
         visit(node.arguments)
       end
 
-      sig { params(node: T.any(SyntaxTree::Def, SyntaxTree::Defs)).void }
+      sig { params(node: SyntaxTree::DefNode).void }
       def add_def_range(node)
+        # For an endless method with no arguments, `node.params` returns `nil` for Ruby 3.0, but a `Syntax::Params`
```

Comment:
@kddnewton is this an expected behaviour? I couldn't see any mention of it in the docs or changelog.

Reply:
Dang yeah I missed that in the changelog, sorry. 

The type in the documentation is right though https://github.com/ruby-syntax-tree/syntax_tree/blame/7e1b2a8176c37786323e334e477da8bd216f6ad6/lib/syntax_tree/node.rb#L4052